examples/statistics/confidence_ellipse.py

```diff
@@ -0,0 +1,222 @@
+    """
+    if x.size != y.size:
+        raise ValueError("x and y must be the same size")
+
+    cov = np.cov(x, y)
+    pearson = cov[0, 1]/np.sqrt(cov[0, 0] * cov[1, 1])
+    # Using a special case to obtain the eigenvalues of this
+    # two-dimensionl dataset.
+    ell_radius_x = np.sqrt(1 + pearson)
+    ell_radius_y = np.sqrt(1 - pearson)
+    ellipse = Ellipse((0, 0),
+        width=ell_radius_x * 2,
+        height=ell_radius_y * 2,
+        facecolor=facecolor,
+        **kwargs)
+
+    # Calculating the stdandard deviation of x from
+    # the squareroot of the variance and multiplying
+    # with the given number of standard deviations.
+    scale_x = np.sqrt(cov[0, 0]) * n_std
+    mean_x = np.mean(x)
+
+    # calculating the stdandard deviation of y ...
+    scale_y = np.sqrt(cov[1, 1]) * n_std
+    mean_y = np.mean(y)
+
+    transf = transforms.Affine2D() \
+        .rotate_deg(45) \
+        .scale(scale_x, scale_y) \
+        .translate(mean_x, mean_y)
+
+    ellipse.set_transform(transf + ax.transData)
+    return ax.add_patch(ellipse)
+
+
+#############################################################################
+#
+# A helper function to create a correlated dataset
+# """"""""""""""""""""""""""""""""""""""""""""""""
+#
+# Creates a random two-dimesional dataset with the specified
+# two-dimensional mean (mu) and dimensions (scale).
+# The correlation can be controlled by the param 'dependency',
+# a 2x2 matrix.
+
+def get_correlated_dataset(n, dependency, mu, scale):
+    latent = np.random.randn(n, 2)
+    dependent = latent.dot(dependency)
+    scaled = dependent * scale
+    scaled_with_offset = scaled + mu
+    # return x and y of the new, correlated dataset
+    return scaled_with_offset[:, 0], scaled_with_offset[:, 1]
+
+
+#############################################################################
+#
+# Positive, negative and weak correlation
+# """""""""""""""""""""""""""""""""""""""
+#
+# Note that the shape for the weak correlation (right) is an ellipse,
+# not a circle because x and y are differently scaled.
+# However, the fact that x and y are uncorrelated is shown by
+# the axes of the ellipse being aligned with the x- and y-axis
+# of the coordinate system.
+
+np.random.seed(0)
+
+PARAMETERS = {
+    'Positive correlation': np.array([[0.85, 0.35],
+                                      [0.15, -0.65]]),
+    'Negative correlation': np.array([[0.9, -0.4],
+                                      [0.1, -0.6]]),
+    'Weak correlation': np.array([[1, 0],
+                                  [0, 1]]),
+}
+
+mu = 2, 4
+scale = np.array([3, 5]).T
+
+fig, axs = plt.subplots(1, 3, figsize=(9, 3))
+for ax, (title, dependency) in zip(axs, PARAMETERS.items()):
+    x, y = get_correlated_dataset(800, dependency, mu, scale)
+    ax.scatter(x, y, s=0.5)
+
+    ax.axvline(c='grey', lw=1)
+    ax.axhline(c='grey', lw=1)
+
+    confidence_ellipse(x, y, ax, facecolor='none', edgecolor='red')
```

Comment:
```suggestion
    confidence_ellipse(x, y, ax, edgecolor='red')
```
note needed anymore.